backend/cfg/cfg.ml

```diff
@@ -424,6 +428,8 @@ let is_pure_operation : operation -> bool = function
   | Compf _ -> true
   | Floatofint -> true
   | Intoffloat -> true
+  | Valueofint -> true
```

Comment:
Shouldn't it be treated like `Opaque` below?

Reply:
I don't think so, I think it is fine to eliminate the expression if the result is not used (similarly to Floatofint/Intoffloat).

Reply:
I thought you wanted to keep such operations
until emission; with this definition, it looks like
the operations might be deleted before register
allocation.

Reply:
I think that was to keep `Valueofint/Intofvalue` until Emit to prevent the "no-op" moves from being eliminated in `Linearize`. (This was also overly-conservative, because liveness of instructions doesn't change after regalloc, at least at the moment, but it was defensive in order to make sure that the regs involved appear correctly in the frametable even if we introduce new passes and need to run liveness again). 

In this case, `is_pure_*` are used  for deadcode elimination, and so the instruction will be deleted only if the result is unused and in particular the `res` is not live, so won't need to be appearing in the frame table.  Am I missing something?

Reply:
Why don't we want the defensive approach here?  This could lead to bugs that are very hard to debug if it goes wrong, no?

Reply:
I don't think so. `Valueofint/Intofvalue` are similar to `Floatofint/Intoffloat`, and both pairs are pure. The difference is that `Floatofint/Intoffloat` would not be eliminated by `linearize` because the `arg` and `res` cannot be the same hard registers (because their type differ), but for `Valueofint/Intofvalue`  can.